### DISCUSSION TITLE
@types/react types missmatches in useAsyncList

### DISCUSSION THREAD

**Discussion by RajaTheKing826** [adobe/react-spectrum#6599] [Q&A]
I've installed packages with the versions you mentioned in may1st release, but still im getting ts errors in **useAsyncList** hook.
Can you tell me what i'm missing?

This is a screenshot from my repo with your code
![image](https://github.com/adobe/react-spectrum/assets/59301270/3c16ef17-2d52-446b-bff0-82ce861df69f)

This is screenshot from your repo
![Image Pasted at 2024-6-24 11-02](https://github.com/adobe/react-spectrum/assets/59301270/b36efd9a-a290-496d-905f-4e2c67f97ad5)

package versions
react:"18.1.0"
@types/react:"18.2.28",

I Dont know why you are not getting errors.
NOTE: above packages versions are as same as your repo on may1st release source code.

**Comment by snowystinger** [comment]
That looks like a react 19 types error... which is odd given that you're on react 18 and we haven't released a version built against react 19 types yet.

Two things to quickly rule out

Can you verify that you have no duplicates of our packages installed. They can be tricky to find, typically a node_modules directory inside another node_modules directory. They can also appear in the lock file if you know how to read them.

Can you triple check the react version and types version and make sure there aren't duplicates of that as well.

**Comment by RajaTheKing826** [comment]
I've cross checked the packages and their versions i have only single version of your packages and react, types/react but i'm still getting the error, i kept packages in resolutions too

```
"@react-stately/collections": "3.10.6",
      "@react-stately/layout": "3.13.8",
      "@react-stately/list": "3.10.4",
      "@react-stately/overlays": "3.6.6",
      "@react-aria/datepicker": "3.10.0",
      "@react-aria/listbox": "3.12.0",
      "@react-aria/overlays": "3.22.0",
      "@react-aria/separator": "3.3.12",
      "@react-aria/tabs": "3.9.0",
      "@react-aria/textfield": "3.14.4",
      "@react-aria/utils": "3.24.0",
      "@react-aria/virtualizer": "3.10.0"
```

These are the packages i'm using. I've kept them in resolutions too for checking.

**Reply by snowystinger** [reply]
Out of curiosity, are you copying our code into your source? If so, we recently started building against react 19 types, so either you can change the TS for that line to be the old way or you can use react 19 types yourself.
See [React 19 support](https://github.com/adobe/react-spectrum/pull/6437#top)

specifically useAsyncList https://github.com/adobe/react-spectrum/pull/6437/files#diff-deb7213ce7d697896860a8e7bc1206037e0fee2ba95707b0856981b8cc7484a9
